zkvm/src/ops.rs

```diff
@@ -187,4 +189,9 @@ impl Instruction {
             Opcode::Delegate => Ok(Instruction::Delegate),
         }
     }
+
+    /// Returns the bytecode representation of an Instruction.
+    pub fn encode(&self) -> Vec<u8> {
```

Comment:
Maybe make it like `write_` APIs, write by appending to the `&mut Vec<u8>` provided as an argument?